### PR TITLE
Layout Flags and Mobile settings views

### PR DESCRIPTION
**Pull request by lorenjohnson**

### CODE REVIEW

**Comment by lorenjohnson** [comment]
So @tibetsprague thank you for reviewing that Mobile Settings Layout PR on Evo last week! 

Unfortunately I should have flagged it as “draft” as it was still in fact a WIP.

Most of what you saw at the time you reviewed, the basic things related to the layout remains unchanged, but since you reviewed I’ve done substantial work on that branch on the Group Settings UI.

It’s ready for another review now. I'm hoping to get this release to coordinate with a Mobile release I'd like to start to QA against production sometime before Sprint Review.

Flagging @thomasgwatson as well in case you have time and a 2nd pass review might be for more benefit to you since Tibet already looked at an earlier version 🤷🏼‍♂️

Thank you!

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@ config/ssl
 .env
 .env.old
 .env.production
+.env.dev
```
Comment (tibetsprague) [inline .gitignore:23]: cant we do .env* or something?

File: src/components/Pillbox/Pillbox.js [hunk 2]
```diff
@@ -134,12 +135,14 @@ export default class Pillbox extends Component {
           ref={this.list} />
         }
       </div>}
-      <ReactTooltip place='top'
-        type='dark'
-        id='pill-label'
-        effect='solid'
-        disable={!editable}
-        delayShow={500} />
+      {!isMobile.any && (
```
Comment (tibetsprague) [inline src/components/Pillbox/Pillbox.js:138]: i wonder if we should just turn off all tooltips on mobile through a custom Tooltip component?

File: src/routes/FullPageModal/FullPageModal.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ export default function FullPageModal ({
 
   if (mobileSettingsLayout) {
     return (
-      <div styleName='modalMobileSettings'>
+      <div styleName='modal--modal-settings-layout'>
```
Comment (tibetsprague) [inline src/routes/FullPageModal/FullPageModal.js:31]: ah the everlasting battle between camels and dashed continues... reasons for a preference for dashes?

File: src/routes/GroupSettings/InviteSettingsTab/InviteSettingsTab.js [hunk 4]
```diff
@@ -127,89 +127,113 @@ ${props.group.name} is using Hylo for our online group: this is our dedicated sp
...
-            small>
-            {reset
-              ? 'Reset'
-              : (inviteLink ? 'Reset Link' : 'New Link')}
-          </Button>
-          {inviteLink && <CopyToClipboard text={inviteLink} onCopy={onCopy}>
-            <Button color={buttonColor(copied)} styleName='styles.copy-button' narrow small>
-              {copied ? 'Copied' : 'Copy Link'}
+
+      {!pending && <>
```
Comment (tibetsprague) [inline src/routes/GroupSettings/InviteSettingsTab/InviteSettingsTab.js:137]: what is this <>?

File: src/routes/GroupSettings/InviteSettingsTab/InviteSettingsTab.scss [hunk 5]
```diff
@@ -201,26 +201,12 @@ $rowHeight: 60px;
...
 
 @media screen and (max-width: 525px) {
   .invite-link-settings {
-    display: block;
-    text-align: center;
-  }
-
-  .switch {
-    margin-left: 20px;
+    // display: block;
```
Comment (tibetsprague) [inline src/routes/GroupSettings/InviteSettingsTab/InviteSettingsTab.scss:209]: should take this out if not needed

File: src/routes/GroupSettings/RelatedGroupsTab/RelatedGroupsTab.js [hunk 6]
```diff
@@ -290,13 +289,13 @@ export function GroupCard ({ actionMenu, thisGroup, group, questionAnswers, type
...
         <div styleName='answer' key={q.id}>
-          <h3>{q.text}</h3>
+          <div styleName='subtitle'>{q.text}</div>
           <p>{get('answer', questionAnswers && questionAnswers.find(qa => qa.question.id === q.questionId)) || <i>Not answered</i>}</p>
         </div>
       )}
       {otherAnswers.map(qa =>
         <div styleName='answer' key={qa.id}>
-          <h3>{qa.question.text}</h3>
+          <div styleName='subtitle'>{qa.question.text}</div>
```
Comment (tibetsprague) [inline src/routes/GroupSettings/RelatedGroupsTab/RelatedGroupsTab.js:298]: why the preference for the styleName over h3?

File: src/routes/GroupSettings/RelatedGroupsTab/RelatedGroupsTab.scss [hunk 7]
```diff
@@ -1,15 +1,29 @@
 .container {
   padding-bottom: 200px;
 
-  h1 {
-    margin-bottom: 0;
-  }
+  // h1 {
```
Comment (tibetsprague) [inline src/routes/GroupSettings/RelatedGroupsTab/RelatedGroupsTab.scss:4]: take out stuff that is no longer needed

File: .gitignore (hunk 1 above)
Comment (lorenjohnson) [reply .gitignore:23]: Removing for now, the thinking here was part of another thought.

File: src/components/Pillbox/Pillbox.js (hunk 2 above)
Comment (lorenjohnson) [reply src/components/Pillbox/Pillbox.js:138]: Maybe, probably. Don't want to take that on as part of this scope right now.

File: src/routes/FullPageModal/FullPageModal.js (hunk 3 above)
Comment (lorenjohnson) [reply src/routes/FullPageModal/FullPageModal.js:31]: indeed. SCSS/SASS is is essential a CSS format and by historical convention dashes and underscores were a thing. We started by sticking with that convention and I prefer it as it delineates between imported CSS styles vs Javascript objects styles (which we may also use or end-up switching to at times). I haven't done a recent review of this in Evo but I suspect that the majority of the CSS classes are named using dashes and underscores. It works for me, and I care very little until it's consistent and we've all decided so happy to go either way. Leaving it for now.

File: src/routes/GroupSettings/InviteSettingsTab/InviteSettingsTab.js (hunk 4 above)
Comment (lorenjohnson) [reply src/routes/GroupSettings/InviteSettingsTab/InviteSettingsTab.js:137]: https://reactjs.org/docs/fragments.html#short-syntax

File: src/routes/GroupSettings/RelatedGroupsTab/RelatedGroupsTab.js (hunk 6 above)
Comment (lorenjohnson) [reply src/routes/GroupSettings/RelatedGroupsTab/RelatedGroupsTab.js:298]: Our reset / stylesheet is not consistently applied for the standard HTML tags. I prefer them, but as they are globally applied and we are using modules in moving towards normalizing this (and User) settings areas I think for now we should be moving away from them and towards a shared module stylesheet across all these screens. Right now it's somewhere 1/2 between. It was more chaotic before I got in there this round :) Moving in increments. Hope that makes sense. In short, it's a little bit of a mess and moving away from the `hX` headers for now is likely to make things a little more clear overall than less. IMHO

**Review by tibetsprague — APPROVED** [review]

**Comment by lorenjohnson** [comment]
Thanks for getting right on that @tibetsprague ! Updates and notes made 👍🏼

**Pull request merged**